src/main/App.ts

```diff
@@ -50,10 +51,16 @@ export class App {
 
       this.redisClient = await this.createRedisClient();
       this.firebaseAdmin = await initFirebaseAdmin();
+      const geocodingService = new GeocodingService({
+        provider: 'locationiq',
+        apiKey: process.env.GEOCODING_KEY,
+      });
+
```

Comment:
Maybe rather than passing in the actual service class, we can pass in the node geocoder instance!

```suggestion
      const nodeGeoCoder = node_geocoder({
        provider: 'locationiq',
        apiKey: process.env.GEOCODING_KEY,
      })
```

Then `GeocodingService's` constructor will take in a parameter of type `Geocoder` rather than of type `Options`. 
And then also `createGeocodingServiceMock` could be `createMockNodeGeocoder` which returns type `Geocoder`.

Reply:
The reason is say in the future imagine we have another service class that needs `node-geocoder`, we would need to add in more parameters.

Reply:
And also like `createGeocodingServiceMock` suggests that we are mocking the service class, but actually, we are mocking `node-geocoder`, our service class isn't being mocked.

Reply:
Oooh yeah those are good points!! Sure I can update them :)